lib/src/protocol/models/rfq.dart

```diff
@@ -41,35 +48,208 @@ class Rfq extends Message {
       externalId: externalId,
     );
 
+    var result = hashPrivateData(createRfqData);
+
     return Rfq._(
       metadata: metadata,
-      data: data,
+      data: result['data'],
+      privateData: result['privateData'],
     );
   }
 
-  static Future<Rfq> parse(String rawMessage) async {
-    final rfq = Parser.parseMessage(rawMessage) as Rfq;
+  static String digestPrivateData(String salt, value) {
+    var digest = CryptoUtils.digest([salt, value]);
+    return Base64Url.encode(digest);
+  }
+
+  static Map<String, dynamic> hashPrivateData(CreateRfqData unhashedRfqData) {
+    var salt = CryptoUtils.generateSalt();
+
+    PrivatePaymentDetails? privatePayin;
+    PrivatePaymentDetails? privatePayout;
+    String? payinDetailsHash;
+    String? payoutDetailsHash;
+    String? claimsHash;
+
+    if (unhashedRfqData.payin.paymentDetails != null) {
+      privatePayin = PrivatePaymentDetails(paymentDetails: unhashedRfqData.payin.paymentDetails!);
+      payinDetailsHash = digestPrivateData(salt, unhashedRfqData.payin.paymentDetails! as Object);
+    }
+
+    if (unhashedRfqData.payout.paymentDetails != null) {
+      privatePayout = PrivatePaymentDetails(paymentDetails: unhashedRfqData.payout.paymentDetails!);
+      payoutDetailsHash = digestPrivateData(salt, unhashedRfqData.payout.paymentDetails! as Object);
+
+    }
+
+    if (unhashedRfqData.claims != null && unhashedRfqData.claims != null) {
+      claimsHash = digestPrivateData(salt, unhashedRfqData.claims);
+    }
+
+    var data = RfqData(
+      offeringId: unhashedRfqData.offeringId,
+      payin: SelectedPayinMethod(
+        amount: unhashedRfqData.payin.amount,
+        kind: unhashedRfqData.payin.kind,
+        paymentDetailsHash: payinDetailsHash,
+      ),
+      payout: SelectedPayoutMethod(
+        kind: unhashedRfqData.payout.kind,
+        paymentDetailsHash: payoutDetailsHash,
+      ),
+      claimsHash: claimsHash,
+    );
+
+
+    var privateData = RfqPrivateData(
+      salt: salt,
+      claims: unhashedRfqData.claims,
+      payin: privatePayin,
+      payout: privatePayout,
+    );
+
+    return {
+      'data': data,
+      'privateData': privateData,
+    };
+  }
+
+  static Future<Rfq> parse(String rawMessage, { requireAllPrivateData = false }) async {
```

Comment:
I'd prefer to keep the `requireAllPrivateData` functionality. My understanding for tbdex-swift was that we omitted that functionality because we were up against a deadline and omitting it from swift was a good trade off to cut down on eng work.

Since I've already paid the upfront cost of implementing this is dart, I don't see any drawback to keeping it in. Even though we won't necessarily use that functionality for didpay, other package consumers might find it useful for their dart apps. E.g. someone writes p2p tbdex client in dart that has both pfi and wallet functionality.